client/abstraction.go

```diff
@@ -0,0 +1,133 @@
+package client
+
+import (
+	"context"
+
+	"github.com/nspcc-dev/neofs-sdk-go/accounting"
+	"github.com/nspcc-dev/neofs-sdk-go/container"
+	cid "github.com/nspcc-dev/neofs-sdk-go/container/id"
+	"github.com/nspcc-dev/neofs-sdk-go/eacl"
+	"github.com/nspcc-dev/neofs-sdk-go/netmap"
+	oid "github.com/nspcc-dev/neofs-sdk-go/object/id"
+	"github.com/nspcc-dev/neofs-sdk-go/reputation"
+	"github.com/nspcc-dev/neofs-sdk-go/user"
+)
+
+// AccountingAPI gives API methods to work with accounting data.
+type AccountingAPI interface {
+	// BalanceGet requests current balance of the NeoFS account.
+	//
+	// See details in [client.Client.BalanceGet].
```

Comment:
it seems weird to me when interface references to particular implementation. I would suggest defining these interfaces in place in packages

Reply:
These just shouldn't be present in `client`, yes.

Reply:
Agree, comments in wrong place. I don't like this variant too. I didn't want to copy-paste all of them to interface definition.
I may just move all comments from `Client` to Interfaces. It may solve problem - definition in one common place.

> it seems weird to me when interface references to particular implementation. I would suggest defining these interfaces in place in packages

Do you mean we should remove these interfaces and they will be defined when needed? I think this isn't a good approach. In this way it will be difficult to keep them synchronous.

> These just shouldn't be present in `client`, yes.

Interfaces belong to owners, not to their implementations - it's ok. Despite it we still have to have them to be sure `Client` and `Pool` may be replaced smoothly. Having them in `client` package looks like a good place.

Reply:
> I may just move all comments from Client to Interfaces.

No, please!

We just need to define whatever interfaces we need for higher-level things wherever these things are implemented. Defining interface and implementation in the same package is not very useful in 95% of the cases. So for now we can just omit these interfaces at all, they're to be defined when they're needed.